mantidimaging/core/io/saver.py

```diff
@@ -289,6 +288,24 @@ def _save_recon_to_nexus(nexus_file: h5py.File, recon: ImageStack):
     data.create_dataset("data", shape=recon.data.shape, dtype="uint16")
     data["data"][:] = _rescale_recon_data(recon.data)
 
+    x_arr, y_arr, z_arr = _create_pixel_size_arrays(recon)
+    data.create_dataset("x", shape=x_arr.shape, dtype="float16", data=x_arr)
+    data.create_dataset("y", shape=y_arr.shape, dtype="float16", data=y_arr)
+    data.create_dataset("z", shape=z_arr.shape, dtype="float16", data=z_arr)
+
+
+def _create_pixel_size_arrays(recon: ImageStack) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Create the pixel size arrays for the NXtomproc x/y/z fields.
+    :param recon: The recon data.
+    :return: The tuple of the x/y/z arrays.
+    """
+    pixel_size = recon.pixel_size
+    x_arr = np.array([pixel_size * (1 + i) for i in range(recon.data.shape[0])])
```

Comment:
I miss read the slack message earlier and gave the wrong answer. I think this would be better starting at zero.

Also it could be written with numpy.arange
`x_arr = np.arange(recon.data.shape[0]) * pixel_size`